src/Services.jl

```diff
@@ -1385,11 +1385,21 @@ gamelift(operation, args=[]) =
 gamelift(a...; b...) = gamelift(a..., b)
 
 function glacier(aws::AWSConfig, verb, resource, args=[])
+    version = "2012-06-01"
+    glacier_version_header = Dict("headers"=>Dict("x-amz-glacier-version"=>version))
+
+    args = Dict(args)
+
+    if haskey(args, "headers")
+        args["headers"] = merge(+, args["headers"], glacier_version_header["headers"])
+    else
+        args = merge(args, glacier_version_header)
+    end
```

Comment:
```suggestion
    glacier_version_header = Dict("x-amz-glacier-version" => version)

    if haskey(args, "headers")
        args["headers"] = merge!(Dict(args["headers"]), glacier_version_header)
    else
        args["headers"] = glacier_version_header
    end
```
Headers doesn't have to be a dict either apparently. Also I don't think `+` does anything good there.